### PR TITLE
[CIS-211] Database flushing adjustments

### PR DESCRIPTION
**Pull request by mantergo**
**In this PR:** 

- Add flushing on model conflicts
- Add option to flush DB on start for SDK users

_Not implemented (details will be in JIRA ticket):  DB flush on ExtraData changes._

### CODE REVIEW

File: Sources_v3/ChatClient.swift [hunk 1]
```diff
@@ -223,6 +223,13 @@ public class _ChatClient<ExtraData: ExtraDataTypes> {
         connectionIdWaiters.removeAll()
     }
     
+    /// Removes all data from the local storage.
+    ///
+    /// - Warning: ⚠️ This is a non-recoverable operation. All data will be lost after calling this method.
+    public func flushLocalStorage(completion: ((Error?) -> Void)? = nil) {
+        databaseContainer.removeAllData(force: true, completion: completion)
+    }
+
```
Comment (VojtaStavik) [inline Sources_v3/ChatClient.swift:232]: It seems we don't need this

File: Sources_v3/Database/DatabaseContainer.swift [hunk 2]
```diff
@@ -189,6 +181,28 @@ class DatabaseContainer: NSPersistentContainer {
...
+    private func loadPersistentStores() throws {
+        var storeLoadingError: Error?
+        
+        loadPersistentStores { _, error in
+            storeLoadingError = error
+        }
+        
+        if storeLoadingError != nil,
+            let url = persistentStoreDescriptions.first!.url {
+            try persistentStoreCoordinator.destroyPersistentStore(at: url, ofType: "sqlite", options: nil)
```
[the remark this answers is not in the document]
Comment (mantergo) [reply Sources_v3/Database/DatabaseContainer.swift:195]: Fixed

File: Sources_v3/Database/DatabaseContainer.swift [hunk 3]
```diff
@@ -54,7 +54,7 @@ class DatabaseContainer: NSPersistentContainer {
     ///   - completion: Called when the container finishes its initialization. If the initialization fails, called
     ///   with an error.
     ///
-    init(kind: Kind, modelName: String = "StreamChatModel", bundle: Bundle? = nil) throws {
+    init(kind: Kind, shouldFlushOnStart: Bool = false, modelName: String = "StreamChatModel", bundle: Bundle? = nil) throws {
```
Comment (VojtaStavik) [inline Sources_v3/Database/DatabaseContainer.swift:57]: Please add `shouldFlushOnStart` to the documentation comment

File: Sources_v3/Database/DatabaseContainer.swift [hunk 4]
```diff
@@ -189,6 +181,29 @@ class DatabaseContainer: NSPersistentContainer {
...
+    
+    /// Load persistent store and recreate it if failed
+    private func setupPersistentStore(shouldFlushOnStart: Bool) throws {
+        var storeLoadingError: Error?
+        
+        loadPersistentStores { _, error in
+            storeLoadingError = error
+        }
+        
+        if (storeLoadingError != nil || shouldFlushOnStart),
```
Comment (VojtaStavik) [inline Sources_v3/Database/DatabaseContainer.swift:193]: If `shouldFlushOnStart == true` you load a persistent store just to delete it on the next line :)

**Review by VojtaStavik — APPROVED** [review]
LGTM 👍 Just some minor comments

**Pull request merged**